examples/optional_redirect/src/tests.rs

```diff
@@ -0,0 +1,27 @@
+use super::rocket;
+use rocket::testing::MockRequest;
+use rocket::http::{Method, Status};
+
+fn test(uri: &str, expected_status: Status, expected_body: Option<String>) {
+    let rocket = rocket::ignite()
+        .mount("/", routes![super::root, super::user, super::login]);
+    let mut request = MockRequest::new(Method::Get, uri);
+    let mut response = request.dispatch_with(&rocket);
+
+    assert_eq!(response.status(), expected_status);
+    assert_eq!(response.body().and_then(|b| b.into_string()), expected_body);
+}
+
+#[test]
+fn test_200() {
+    test("/users/Sergio", Status::Ok, Some("Hello, Sergio!".to_string()));
+    test("/users/login",
+         Status::Ok,
+         Some("Hi! That user doesn't exist. Maybe you need to log in?".to_string()));
+}
+
+#[test]
+fn test_303() {
```

Comment:
Good catch, I didn't think to test the Location header. Tests have been updated!